### PR TITLE
Nlb udp

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
--- PASS: TestAccIBMISLB_basic_udp (713.17s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     714.768s
```
```
--- PASS: TestAccIBMISLBPool_basic_udp (802.21s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     803.506s
```

```
--- PASS: TestAccIBMISLBListener_basic_udp (794.38s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     795.690s
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_lb.go [hunk 1]
```diff
@@ -469,6 +476,9 @@ func lbGet(d *schema.ResourceData, meta interface{}, id string) error {
 
 	d.Set(isLBResourceGroup, *lb.ResourceGroup.ID)
 	d.Set(isLBHostName, *lb.Hostname)
+	if lb.UDPSupported != nil {
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_lb.go:479]: isLBUdpSupported is supported only for network load balancer hence i feel there should also be optional: true.
Please reverify 
https://pages.github.ibm.com/vpc/vpc-spec-artifacts/branch/master/public/development-tomorrow.html#/Load%20balancers/create_load_balancer

File: ibm/service/vpc/resource_ibm_is_lb.go [hunk 2]
```diff
@@ -175,6 +176,12 @@ func ResourceIBMISLB() *schema.Resource {
 				Description: "Indicates whether route mode is enabled for this load balancer",
 			},
 
+			isLBUdpSupported: {
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_lb.go:179]: Please add acceptance test

File: ibm/service/vpc/resource_ibm_is_lb_listener.go [hunk 3]
```diff
@@ -176,7 +176,7 @@ func ResourceIBMISLBListener() *schema.Resource {
 func ResourceIBMISLBListenerValidator() *validate.ResourceValidator {
 
 	validateSchema := make([]validate.ValidateSchema, 0)
-	protocol := "https, http, tcp"
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_lb_listener.go:179]: Please add acceptance test

File: ibm/service/vpc/resource_ibm_is_lb_pool.go [hunk 4]
```diff
@@ -186,7 +186,7 @@ func ResourceIBMISLBPoolValidator() *validate.ResourceValidator {
 
 	validateSchema := make([]validate.ValidateSchema, 0)
 	algorithm := "round_robin, weighted_round_robin, least_connections"
-	protocol := "http, tcp, https"
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_lb_pool.go:189]: Please add acceptance test

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hi Ujjwal, 
I have reviewed your code, please have a look..
Thanks, 
Sunitha

File: ibm/service/vpc/data_source_ibm_is_lbs.go [hunk 5]
```diff
@@ -56,6 +56,11 @@ func DataSourceIBMISLBS() *schema.Resource {
 							Computed:    true,
 							Description: "Load Balancer name",
 						},
+						isLBUdpSupported: {
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/data_source_ibm_is_lbs.go:59]: isLBUdpSupported is applicable only for network load balancer, hence i think optional: true should also be there, Please reverify

**Comment by SunithaGudisagarIBM** [comment]
Hi Team,

I have tested this PR with the below config and it is working as expected.

resource "ibm_is_lb" "testacc_LB" {
  name    = "sunitha"
  subnets = ["0716-221dbf30-583f-4d13-9140-80aaaa1b09c1"]
}

resource "ibm_is_lb" "testacc_NLB" {
  name    = "sunitha-nw"
  subnets = ["0716-221dbf30-583f-4d13-9140-80aaaa1b09c1"]
  profile = "network-fixed"
}

resource "ibm_is_lb_pool" "testacc_lb_pool_nlb" {
  name               = "sunitha-pool"
  lb                 = ibm_is_lb.testacc_NLB.id
  algorithm          = "round_robin"
  protocol           = "udp"
  health_delay       = "5"
  health_retries     = "2"
  health_timeout     = "2"
  health_type        = "http"
  health_monitor_url = "/"
}

resource "ibm_is_lb_pool" "testacc_lb_pool_lb" {
  name           = "sunitha-pool-lb"
  lb             = ibm_is_lb.testacc_LB.id
  algorithm      = "round_robin"
  protocol       = "http"
  health_delay   = "5"
  health_retries = "2"
  health_timeout = "2"
  health_type    = "http"
}

resource "ibm_is_lb_listener" "testacc_lb_listener_nlb" {
  lb       = ibm_is_lb.testacc_NLB.id
  port     = 8080
  protocol = "udp"
}


resource "ibm_is_lb_listener" "testacc_lb_lister_lb" {
  lb       = ibm_is_lb.testacc_LB.id
  port     = 8080
  protocol = "http"
}

data "ibm_is_lb" "ds_lb" {
  name = ibm_is_lb.testacc_LB.name
}


data "ibm_is_lb" "ds_nlb" {
  name = ibm_is_lb.testacc_NLB.name
}
data "ibm_is_lbs" "test_lbs" {
}

data "ibm_is_lb_pools" "is_lb_pools" {
  lb = ibm_is_lb.testacc_LB.id
}

data "ibm_is_lb_pools" "is_nlb_pools" {
  lb = ibm_is_lb.testacc_NLB.id
}

data "ibm_is_lb_pool" "is_lb_pool" {
  lb         = ibm_is_lb.testacc_LB.id
  identifier = element(split("/", ibm_is_lb_pool.testacc_lb_pool_lb.id), 1)
}

data "ibm_is_lb_pool" "is_nlb_pool" {
  lb         = ibm_is_lb.testacc_NLB.id
  identifier = element(split("/", ibm_is_lb_pool.testacc_lb_pool_nlb.id), 1)
}

data "ibm_is_lb_listeners" "is_lb_listeners" {
  lb = ibm_is_lb.testacc_LB.id
}

data "ibm_is_lb_listeners" "is_nlb_listeners" {
  lb = ibm_is_lb.testacc_NLB.id
}

data "ibm_is_lb_listener" "is_lb_listener" {
  lb          = ibm_is_lb.testacc_LB.id
  listener_id = ibm_is_lb_listener.testacc_lb_lister_lb.listener_id
}

data "ibm_is_lb_listener" "is_nlb_listener" {
  lb          = ibm_is_lb.testacc_NLB.id
  listener_id = ibm_is_lb_listener.testacc_lb_listener_nlb.listener_id
}



Screenshot 2022-03-23 at 3 31 34 PM
Screenshot 2022-03-23 at 3 31 51 PM
Screenshot 2022-03-23 at 3 32 04 PM
Screenshot 2022-03-23 at 3 32 16 PM
Screenshot 2022-03-23 at 3 32 32 PM
Screenshot 2022-03-23 at 3 33 02 PM
Screenshot 2022-03-23 at 3 33 18 PM

**Comment by SunithaGudisagarIBM** [comment]
When tested with master branch binary and this is expected.. 

Screenshot 2022-03-23 at 3 35 52 PM

**Comment by ujjwal-ibm** [comment]
moving to upstream

**Pull request closed without merging**